src/tools/genpolicy/rules.rego

```diff
@@ -457,16 +468,93 @@ allow_log_directory(p_oci, i_oci) if {
     print("allow_log_directory: true")
 }
 
-allow_devices(p_devices, i_devices) if {
+allow_devices(p_devices, i_devices, i_oci) if {
     print("allow_devices: start")
-    every i_device in i_devices {
-        print("allow_devices: i_device =", i_device)
-        some p_device in p_devices
-        p_device.container_path == i_device.container_path
-    }
+
+    vfio_device_path := policy_data.device_annotations.vfio.device_path
+
+    p_volume_devices := [d | d := p_devices[_]; d.container_path != vfio_device_path]
+    i_volume_devices := [d | d := i_devices[_]; not startswith(d.container_path, vfio_device_path)]
+    allow_volume_devices(p_volume_devices, i_volume_devices)
```

Comment:
Print the two collections after calculating them - here and for vfio devices below.